execute “scons -u -j$(nproc)” error on MacOS with M1 Pro

### Describe the bug

openpilot tools

tools/mac_setup.sh    ok
pipenv shell     ok
scons -u -j$(nproc)     error
-------

clang -o selfdrive/controls/lib/lateral_mpc_lib/c_generated_code/acados_ocp_solver_pyx.so -bundle -undefined dynamic_lookup /Users/maxingyuan/Documents/code/autodriver/0521/openpilot/selfdrive/controls/lib/lateral_mpc_lib/c_generated_code/libacados_ocp_solver_lat.dylib selfdrive/controls/lib/lateral_mpc_lib/c_generated_code/acados_ocp_solver_pyx.o -Lthird_party/libyuv/mac_arm64/lib -L/opt/homebrew/lib -L/opt/homebrew/Library -L/opt/homebrew/opt/openssl/lib -L/opt/homebrew/Cellar -Lthird_party/acados/Darwin/lib -L/System/Library/Frameworks/OpenGL.framework/Libraries -Lcereal -Lthird_party -Lopendbc/can -Lselfdrive/boardd -Lcommon
clang++ -o selfdrive/loggerd/encoder/encoder.o -c -std=c++1z -DGL_SILENCE_DEPRECATION -DSWAGLOG="\"common/swaglog.h\"" -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DGL_SILENCE_DEPRECATION -DSWAGLOG="\"common/swaglog.h\"" -I/opt/homebrew/include -I/opt/homebrew/opt/openssl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/curl/include -Ithird_party/libgralloc/include -Ithird_party/android_frameworks_native/include -Ithird_party/android_hardware_libhardware/include -Ithird_party/android_system_core/include -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/mapbox-gl-native-qt/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/json11 selfdrive/loggerd/encoder/encoder.cc
clang++ -o selfdrive/loggerd/encoder/ffmpeg_encoder.o -c -std=c++1z -DGL_SILENCE_DEPRECATION -DSWAGLOG="\"common/swaglog.h\"" -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DGL_SILENCE_DEPRECATION -DSWAGLOG="\"common/swaglog.h\"" -I/opt/homebrew/include -I/opt/homebrew/opt/openssl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/curl/include -Ithird_party/libgralloc/include -Ithird_party/android_frameworks_native/include -Ithird_party/android_hardware_libhardware/include -Ithird_party/android_system_core/include -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/mapbox-gl-native-qt/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/json11 selfdrive/loggerd/encoder/ffmpeg_encoder.cc
selfdrive/loggerd/video_writer.cc:77:7: error: 'av_init_packet' is deprecated [-Werror,-Wdeprecated-declarations]
      av_init_packet(&pkt);
      ^
/opt/homebrew/include/libavcodec/packet.h:505:1: note: 'av_init_packet' has been explicitly marked deprecated here
attribute_deprecated
^
/opt/homebrew/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
1 error generated.
scons: *** [selfdrive/loggerd/video_writer.o] Error 1
scons: building terminated because of errors.

### OS Version

MacOS 12.3.1  Apple M1 Pro

### openpilot version or commit

_No response_

### Additional info

_No response_

--- a/SConstruct
+++ b/SConstruct
@@ -331,13 +331,13 @@ if GetOption("clazy"):
 Export('env', 'qt_env', 'arch', 'real_arch', 'SHARED', 'USE_WEBCAM')
 
 SConscript(['common/SConscript'])
-Import('_common', '_gpucommon', '_gpu_libs')
+Import('_common', '_gpucommon')
 
 if SHARED:
   common, gpucommon = abspath(common), abspath(gpucommon)
 else:
   common = [_common, 'json11']
-  gpucommon = [_gpucommon] + _gpu_libs
+  gpucommon = [_gpucommon]
 
 Export('common', 'gpucommon')
 

--- a/common/SConscript
+++ b/common/SConscript
@@ -22,14 +22,8 @@ files = [
   'visionimg.cc',
 ]
 
-# if arch == "larch64":
-#   _gpu_libs = ["GLESv2"]
-# else:
-#   _gpu_libs = ["GL"]
-_gpu_libs = []
-
-_gpucommon = fxn('gpucommon', files, LIBS=_gpu_libs)
-Export('_common', '_gpucommon', '_gpu_libs')
+_gpucommon = fxn('gpucommon', files)
+Export('_common', '_gpucommon')
 
 if GetOption('test'):
   env.Program('tests/test_util', ['tests/test_util.cc'], LIBS=[_common])

--- a/common/SConscript
+++ b/common/SConscript
@@ -22,10 +22,11 @@ files = [
   'visionimg.cc',
 ]
 
-if arch == "larch64":
-  _gpu_libs = ["GLESv2"]
-else:
-  _gpu_libs = ["GL"]
+# if arch == "larch64":
+#   _gpu_libs = ["GLESv2"]
+# else:
+#   _gpu_libs = ["GL"]
+_gpu_libs = []
 
 _gpucommon = fxn('gpucommon', files, LIBS=_gpu_libs)
 Export('_common', '_gpucommon', '_gpu_libs')

--- a/selfdrive/loggerd/video_writer.cc
+++ b/selfdrive/loggerd/video_writer.cc
@@ -1,3 +1,4 @@
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #include <cassert>
 #include <cstdlib>
 

--- a/selfdrive/modeld/runners/snpemodel.h
+++ b/selfdrive/modeld/runners/snpemodel.h
@@ -1,4 +1,5 @@
 #pragma once
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 
 #include <DlContainer/IDlContainer.hpp>
 #include <DlSystem/DlError.hpp>

